Applications/Spire/Source/TimeAndSales/TimeAndSalesTableView.cpp

```diff
@@ -210,36 +247,20 @@ TableView* Spire::make_time_and_sales_table_view(
     set_header(make_header_model()).
     set_item_builder(std::bind_front(&item_builder, table)).make();
   update_style(*table_view, apply_table_view_style);
-  auto& header = table_view->get_header();
   auto properties = make_header_item_properties();
-  auto table_header_menu = new ContextMenu(header);
+  make_header_menu(*table_view, properties);
+  auto& header = table_view->get_header();
   for(auto i = 0; i < std::ssize(properties); ++i) {
     header.get_widths()->set(i, properties[i].m_width);
-    auto item = header.get_item(i);
-    item->setVisible(properties[i].m_is_visible);
-    update_style(*item, [&] (auto& style) {
+    update_style(*header.get_item(i), [&] (auto& style) {
       style.get(Any() > TableHeaderItem::Label()).
         set(TextAlign(properties[i].m_alignment));
     });
-    auto model =
-      table_header_menu->add_check_box(item->get_model()->get().m_name);
-    model->set(properties[i].m_is_visible);
-    model->connect_update_signal([=] (auto checked) {
-      table_view->get_header().get_item(i)->setVisible(checked);
-    });
   }
-  header.setContextMenuPolicy(Qt::CustomContextMenu);
-  QObject::connect(&header, &QWidget::customContextMenuRequested,
-    [=] (const auto& pos) {
-      table_header_menu->window()->move(
-        table_view->get_header().mapToGlobal(pos));
-      table_header_menu->window()->show();
-    });
   auto body = &table_view->get_body();
-  auto pull_indicator = make_pull_indicator(body);
-  body->installEventFilter(new TableBodyEventFilter(*pull_indicator, body));
-  auto scroll_box = &table_view->get_scroll_box();
+  auto pull_indicator = new PullIndicator(body);
```

Comment:
Don't spread out implementation details. The ```PullIndicator``` is a good start, finish up the implementation and keep all logically similar parts of the code together.